lib/actions/fcs.py

```diff
@@ -27,29 +27,572 @@ def __init__(self, request, ui_lang):
+            data["responsePosition"] = responsePosition
+
+            # set content-type in HTTP header
+            recordPacking = req.args.get("recordPacking", "xml")
+            if recordPacking == "xml":
+                pass
+            elif recordPacking == "string":
+                # TODO(jm)!!!
+                self._headers["Content-Type"] = "text/plain; charset=utf-8"
+            else:
+                raise Exception(71, 'recordPacking', 'Unsupported record packing')
+
+            # provide info about service
+            operation = req.args.get("operation", "explain")
+            data["operation"] = operation
+            if operation == "explain":
+                self._check_args(
+                    req, supported_args,
+                    ['recordPacking', 'x-fcs-endpoint-description']
+                )
+                corpus = self.cm.get_Corpus(corpname)
+                data["result"] = corpus.get_conf('ATTRLIST').split(',')
+                data["numberOfRecords"] = len(data['result'])
+
+            # wordlist for a given attribute
+            elif operation == 'scan':
+                self._check_args(
+                    req, supported_args,
+                    ['scanClause', 'responsePosition', 'maximumTerms', 'x-cmd-resource-info']
+                )
+                data['resourceInfoRequest'] = req.args.get("x-cmd-resource-info", "") == 'true'
+                scanClause = req.args.get("scanClause", "")
+                if scanClause.startswith("fcs.resource="):
+                    value = scanClause.split("=")[1]
+                    data['result'] = self._corpora_info(value, maximumTerms)
+                else:
+                    data['result'] = conclib.fcs_scan(
+                        corpname, scanClause, maximumTerms, responsePosition)
+
+            # simple concordancer
+            elif operation == 'searchRetrieve':
+                self._check_args(
+                    req, supported_args,
+                    ['query', 'startRecord', 'maximumRecords', 'recordPacking',
+                        'recordSchema', 'resultSetTTL', 'x-cmd-context']
+                )
+                if "x-cmd-context" in req.args:
+                    req_corpname = req.args["x-cmd-context"]
+                    user_corpora = plugins.get('auth').permitted_corpora(
+                        self._session_get('user', 'id')).values()
+                    if req_corpname in user_corpora:
+                        corpname = req_corpname
+                query = req.args.get("query", "")
+                corpus = self.cm.get_Corpus(corpname)
+                if 0 == len(query):
+                    raise Exception(7, 'fcs_query', 'Mandatory parameter not supplied')
+                data['result'] = self.fcs_search(
+                    corpus, corpname, query, maximumRecords, startRecord)
+                data['numberOfRecords'] = len(data['result'])
 
-    @exposed(return_type='xml')
-    def scan(self, request):
```

Comment:
at the first glance some endpoints (scan, index ...) are gone; not sure what the specs say (or if they really are gone)

Reply:
those were not according to spec, there is only one endpoint with parameters